packages/components/src/assay/AssayDesignPage.tsx

```diff
@@ -0,0 +1,219 @@
+import { LoadingPage } from '../internal/components/base/LoadingPage';
+import { Page } from '../internal/components/base/Page';
+
+import { Alert } from '../internal/components/base/Alert';
+import { getActionErrorMessage } from '../internal/util/messaging';
+import { AssayDesignerPanels } from '../internal/components/domainproperties/assay/AssayDesignerPanels';
+import { ProductMenuModel } from '../internal/components/navigation/model';
+
+import { DEFAULT_SAMPLE_FIELD_CONFIG } from '../internal/components/samples/constants';
+
+import { protocolHasSample, renderSampleRequiredPanelHeader } from './SampleRequiredDomainHeader';
+import { AssayHeader } from './AssayHeader';
+import { onAssayDesignChange } from './actions';
+import { useAssayAppContext } from './AssayAppContext';
+import { assayPage } from './AssayPageHOC';
+
+const ASSAY_DESIGNER_HEADER = 'Connect your experimental results to samples for rich data connections.';
+
+// Need to do an exclusion list to allow unknown file-based assays to have batch fields
+const REMOVE_BATCH_DOMAIN_ASSAYS = [
+    'General',
+    'ELISA',
+    'ELISpot',
+    'Noblis Simple',
+    'NAb',
+    'Viability',
+    'TZM-bl Neutralization (NAb)',
+    'TZM-bl Neutralization (NAb), High-throughput (Cross Plate Dilution)',
+    'TZM-bl Neutralization (NAb), High-throughput (Single Plate Dilution)',
+];
+
+interface OwnProps {
+    goBack: (n?: number) => void;
+    menu: ProductMenuModel;
+    menuInit: (invalidate?: boolean) => void;
+    navigate: (url: string | AppURL, replace?: boolean) => void;
+    requireSampleField?: boolean;
+    showProviderName?: boolean;
+}
+
+type Props = OwnProps & InjectedAssayModel & WithRouterProps & InjectedRouteLeaveProps;
+
+const AssayDesignPageBody: FC<Props> = memo(props => {
+    const {
+        assayDefinition,
+        assayProtocol,
+        reloadAssays,
+        router,
+        routes,
+        goBack,
+        menu,
+        menuInit,
+        navigate,
+        params,
+    } = props;
+    const [hasError, setHasError] = useState(false);
+    const [protocol, setProtocol] = useState<AssayProtocolModel>(undefined);
+    const { user } = useServerContext();
+    const [_, setIsDirty] = useRouteLeave(router, routes);
+    const { createNotification } = useNotificationsContext();
+    const { requireSampleField, showProviderName } = useAssayAppContext();
+
+    useEffect(() => {
+        async function getProtocol(protocolId, providerName, copy) {
+            try {
+                let protocol = await fetchProtocol(protocolId, providerName, copy);
+                setHasError(false);
+
+                // Update pre-populated fields and batch fields if we aren't copying the assay design
+                if (!assayProtocol) {
+                    let newDomains = List<DomainDesign>();
+                    protocol.domains.forEach(dom => {
+                        let newDomain = dom;
+
+                        // Clear all pre-populated fields for general assay
```

Comment:
comment needs to be updated to clarify both cases of the if statement

Reply:
Done.